runtime/gc_ctrl.c

```diff
@@ -57,33 +57,33 @@ CAMLprim value caml_gc_quick_stat(value v)
   /* get a copy of these before allocating anything... */
   intnat majcoll;
   struct gc_stats s;
-  caml_sample_gc_stats(&s);
+  caml_compute_gc_stats(&s);
   majcoll = Caml_state->stat_major_collections;
```

Comment:
I was thinking about moving the `stat_major_collections` stat at the same level as `minor_collections`, it feels odd that we have this one retrieved directly from `Caml_state` when we do not do the same for `minor_collections`.
Maybe this could be part of the refactor as well?